pkg/operator/ceph/cluster/rbd/mirror.go

```diff
@@ -158,7 +159,13 @@ func (m *Mirroring) Start() error {
 					return fmt.Errorf("failed to recreate rbd-mirror deployment %s during del-and-recreate update attempt. %+v", resourceName, err)
 				}
 			}
-		} else {
+			if existingDeployment, err := m.context.Clientset.AppsV1().Deployments(m.Namespace).Get(d.GetName(), metav1.GetOptions{}); err != nil {
```

Comment:
I think this should be un-nested one level from the if branch. Otherwise, this will not run if `err == nil` for initial `Create` above.

Reply:
You're entirely right: I was drawn in by the logged deployment start below it (which I'm also bringing out one level), but this was still a pure mistake. Sorry for that; thanks for catching it.